### PR TITLE
Migrate to large test framework

### PR DESCRIPTION
**Pull request by nanliu**
* use the new large test framework
* use psutil instead of meminfo to remove linux /proc dependency
* add test coverage for multiple snap versions
* add test coverage for udp/http protocol

### CODE REVIEW

**Comment by nanliu** [comment]
NOTE: this is not expected to run on jenkins.

File: examples/tasks/psutil-influxdb-udp.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        /intel/psutil/load/load15: {}
+        /intel/psutil/load/load5: {}
+        /intel/psutil/vm/available: {}
+        /intel/psutil/vm/free: {}
+        /intel/psutil/vm/used: {}
+      publish:
+        - plugin_name: "influxdb"
+          config:
+             host: "influxdb"
+             port: 8086
```
Comment (candysmurf) [inline examples/tasks/psutil-influxdb-udp.yml]: s/8086/4444

File: scripts/large.sh [hunk 2]
```diff
@@ -10,37 +10,26 @@ __proj_name="$(basename $__proj_dir)"
...
-set -e
-_debug "exit code $test_res"
-_debug "stopping docker compose images"
-_docker_project docker-compose stop
-_debug "removing docker compose images"
-_docker_project docker-compose rm -f
-exit $test_res
\ No newline at end of file
+_info "running large test"
+docker run -v /var/run/docker.sock:/var/run/docker.sock -v "${__proj_dir}":/plugin -e PLUGIN_PATH="${PLUGIN_PATH}" -e SNAP_VERSION="${SNAP_VERSION}" -e OS="${OS}" -ti nanliu/docker_serverspec:alpine /bin/sh -c "${cmd}"
```
Comment (candysmurf) [inline scripts/large.sh]: is "naliu/docker_serverspec:alpine" special for influx?

File: scripts/test/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      - ${PLUGIN_PATH}:/plugin
+    links:
+      - influxdb
+  influxdb:
+    image: tutum/influxdb
+    environment:
+      UDP_DB: "test"
+    expose:
+      - "8083"
+      - "8086"
```
Comment (candysmurf) [inline scripts/test/docker-compose.yml]: add - "4444/udp"  to support UDP

File: scripts/large.sh (hunk 2 above)
Comment (nanliu) [inline scripts/large.sh]: No, it's our general serverspec container. Let me push this to intelsdi repo and I'll update this.

**Comment by nanliu** [comment]
@candysmurf, PR updated

File: examples/tasks/psutil-influxdb-http.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        /intel/psutil/load/load1: {}
+        /intel/psutil/load/load15: {}
+        /intel/psutil/load/load5: {}
+        /intel/psutil/vm/available: {}
+        /intel/psutil/vm/free: {}
+        /intel/psutil/vm/used: {}
+      publish:
+        - plugin_name: "influxdb"
+          config:
+             host: "influxdb"
```
Comment (kindermoumoute) [inline examples/tasks/psutil-influxdb-http.yml]: Can we replace the host to `${INFLUXDB_HOST}`?

File: examples/tasks/psutil-influxdb-udp.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        /intel/psutil/load/load1: {}
+        /intel/psutil/load/load15: {}
+        /intel/psutil/load/load5: {}
+        /intel/psutil/vm/available: {}
+        /intel/psutil/vm/free: {}
+        /intel/psutil/vm/used: {}
+      publish:
+        - plugin_name: "influxdb"
+          config:
+             host: "influxdb"
```
Comment (kindermoumoute) [inline examples/tasks/psutil-influxdb-udp.yml]: Can we replace the host to `${INFLUXDB_HOST}`?

File: scripts/test/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+version: '2'
+services:
+  snap:
+    image: intelsdi/snap:alpine_test
+    environment:
+      SNAP_VERSION: ${SNAP_VERSION}
```
Comment (kindermoumoute) [inline scripts/test/docker-compose.yml]: And then add a new line `INFLUXDB_HOST: influxdb`

**Comment by candysmurf** [comment]
LGTM

**Pull request merged**